torch/_dynamo/symbolic_convert.py

```
@@ -3960,7 +3960,13 @@ def inline_call_(self):
             ):
                 assert isinstance(self, InliningGeneratorInstructionTranslator)
                 # When the generator returns None, we raise StopIteration
-                exc.raise_observed_exception(StopIteration, self)
+                args = []
+                if (
+                    isinstance(self.symbolic_result, ConstantVariable)
+                    and self.symbolic_result.value is not None
+                ):
+                    args = [self.symbolic_result]
```

Should we raise an unimplemented error for the else case?